src/MySQL.php

```diff
@@ -676,7 +676,22 @@ protected function convertDataForImport(array $data, \obo\Carriers\EntityInforma
             $entityInformationForPropertyRepositoryName = &$storageInformation["repositories"][$propertyRepositoryName];
 
             if ($entityInformationForPropertyRepositoryName["columns"][$entityInformationForPropertyRepositoryName["toColumnName"][$propertyName]]["autoIncrement"]) continue;
-            $convertedData[$propertyStorageName][$propertyRepositoryName][$entityInformationForPropertyRepositoryName["toColumnName"][$propertyName]] = ($entityInformationForPropertyRepositoryName["columns"][$entityInformationForPropertyRepositoryName["toColumnName"][$propertyName]]["importFilter"] === null OR $propertyValue === null) ? $propertyValue : $this->dataConverter->{$entityInformationForPropertyRepositoryName["columns"][$entityInformationForPropertyRepositoryName["toColumnName"][$propertyName]]["importFilter"]}($propertyValue, $propertyInformation);
+            $convertedDataTemporary[$propertyStorageName][$propertyRepositoryName][$entityInformationForPropertyRepositoryName["toColumnName"][$propertyName]] = ($entityInformationForPropertyRepositoryName["columns"][$entityInformationForPropertyRepositoryName["toColumnName"][$propertyName]]["importFilter"] === null OR $propertyValue === null) ? $propertyValue : $this->dataConverter->{$entityInformationForPropertyRepositoryName["columns"][$entityInformationForPropertyRepositoryName["toColumnName"][$propertyName]]["importFilter"]}($propertyValue, $propertyInformation);
+        }
+
+        $convertedData = [];
+        $propertyRepositoryNameOfPrimaryProperty = $entityInformation->informationForPropertyWithName($entityInformation->primaryPropertyName)->repositoryName ?: $entityInformation->repositoryName;
```

Comment:
maybe better(in naming convention) name for variable would be "PrimaryPropertyRepositoryName" and "PrimaryPropertyStorageName" below

Reply:
changed